hotsos/core/ycheck/events.py

```diff
@@ -187,7 +191,23 @@ def categorise_events(cls, event, results=None, key_by_date=True,
                 for key in info:
                     info[key] = sorted_dict(info[key])
 
-            return sorted_dict(info, reverse=not key_by_date)
+            results = sorted_dict(info, reverse=not key_by_date)
+            if not (key_by_date and max_results_per_date):
+                return results
+
+            shortened = {}
+            for date, entries in results.items():
+                if len(entries) <= max_results_per_date:
+                    shortened[date] = entries
+                    continue
+
+                top_n = dict(sorted(entries.items(),
+                                    key=lambda e: e[1],
+                                    reverse=True)[:max_results_per_date])
+                shortened[date] = {'total': len(entries),
+                                   "top{}".format(max_results_per_date): top_n}
```

Comment:
Question on this, and maybe it doesn't matter, but will 'total' always be the first element printed? I thought it was indeterminate so could be 'top5' first. Although maybe this changed as of python 3.7 according to google :)

Reply:
I think it is technically not possible to guarantee but seems to be fairly consistently in that order in py38

Reply:
I believe Python 3.7+ guarantees that the dict class will [keep the insertion order](https://docs.python.org/3/whatsnew/3.7.html#:~:text=the%20insertion%2Dorder%20preservation%20nature%20of%20dict%20objects%20has%20been%20declared%20to%20be%20an%20official%20part%20of%20the%20Python%20language%20spec.). I'd expect yaml output to preserve the order if the *_dump function is being called with `sort_keys=False`, or alternatively we can prepend a `_` to the total if the `sort_keys=True`